hw/ip/otbn/dv/rig/rig/gens/bad_load_store.py

```diff
@@ -0,0 +1,284 @@
+        # SW has three operands: grs1, offset, grs2
+        if not (len(self.sw.operands) == 3 and
+                isinstance(self.sw.operands[0].op_type, RegOperandType) and
+                self.sw.operands[0].op_type.reg_type == 'gpr' and
+                not self.sw.operands[0].op_type.is_dest() and
+                isinstance(self.sw.operands[1].op_type, ImmOperandType) and
+                isinstance(self.sw.operands[2].op_type, RegOperandType) and
+                self.sw.operands[2].op_type.reg_type == 'gpr' and
+                not self.sw.operands[2].op_type.is_dest()):
+            raise RuntimeError('SW instruction from instructions file is '
+                               'not the shape expected by the BadLoadStore'
+                               'generator.')
+
+        # Determine the offset range for XW
+        offset_rng = imm_op_type.get_op_val_range(model.pc)
+        assert offset_rng is not None
+
+        # Max/Min Offsets for XW (-2048, 2047)
+        min_offset, max_offset = offset_rng
+
+        # Get known registers
+        known_regs = model.regs_with_known_vals('gpr')
+
+        # Pick a random register among known registers
+        idx, value = random.choice(known_regs)
+
+        # If it is close enough, pick top of the memory for "barely" out of bound
+        # coverage point for XW instructions
+        if (model.dmem_size + min_offset < value and  # value > 4096 + (-2048)
+           value <= model.dmem_size - min_offset):  # value <= 4096 - (-2048)
+            imm_val = model.dmem_size - value
+        else:
+            imm_val = random.randrange(min_offset, max_offset)
+
+        if value >> 31:
+            addr = imm_val + (value - (1 << 32))
+        else:
+            addr = imm_val + value
+
+        while not (addr < 0 or model.dmem_size <= addr):
+            imm_val = random.randrange(min_offset, max_offset)
+            if value >> 31:
+                addr = imm_val + (value - (1 << 32))
+            else:
+                addr = imm_val + value
+
+        offset_val = imm_op_type.op_val_to_enc_val(imm_val, model.pc)
+        assert offset_val is not None
+
+        # Get the chosen base register index as grs1 operand.
+        op_val_grs1 = idx
+        assert op_val_grs1 is not None
+
+        if insn.mnemonic == 'lw':
+            # Pick grd randomly. Since we can write to any register
+            # (including x0) this should always succeed.
+            op_val_grd = model.pick_operand_value(grd_op_type)
+            assert op_val_grd is not None
+
+            op_val = [op_val_grd, offset_val, op_val_grs1]
+
+        elif insn.mnemonic == 'sw':
+            # Any known register is okay for grs2 operand since it will be
+            # guaranteed to have an out of bounds address to store the value from
+            # We definitely have a known register (x0), so this should never fail
+            op_val_grs2 = random.choice(known_regs)[0]
+            assert op_val_grs2 is not None
```

Comment:
Nit: This assertion isn't really needed. If you want to assert that this won't go wrong, you could put `assert known_regs` on the line before, but you don't need to do so to guide mypy (since `random.choice` doesn't return `None` on failure).

Similarly for bn.sid below.